Duke.java

```diff
@@ -0,0 +1,185 @@
+        System.out.println("added: " + task.name);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    // Method to list tasks
+    public static void listTasks() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCounter; ++i) {
+            Task task = taskList[i];
+            System.out.println(task.toString());
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    // Method to mark a task as done
+    public static void markTask(Task task) {
+        System.out.println(HORIZONTAL_LINE);
+        task.isDone = true;
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("[" + task.getStatusIcon() + "] " + task.name);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    // Method to unmark task as undone
+    public static void unmarkTask(Task task) {
+        System.out.println(HORIZONTAL_LINE);
+        task.isDone = false;
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println("[" + task.getStatusIcon() + "] " + task.name);
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    // Method to add todo, deadline and event tasks
+    public static void addTaskSpecial(String name, String type) {
+        String divider = "/";
+        if (!checkSpace(name)) {
+            errorMessage();
+            return;
+        }
+
+        if (!name.contains(divider)) {
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Please add a date");
+            System.out.println(HORIZONTAL_LINE);
+            greet();
+            return;
+        }
+
+        name = name.substring(name.indexOf(" "), name.length());
+        switch (type) {
+        case "todo":
+            Todo todoTask = new Todo(name);
+            taskList[taskCounter] = todoTask;
+            taskCounter += 1;
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(todoTask.toString());
+            System.out.println("Now you have " + taskCounter + " tasks in the list.");
+            System.out.println(HORIZONTAL_LINE);
+            break;
+        case "deadline":
+            String deadlineDate = name.substring(name.indexOf(divider) + 3, name.length());
+            Deadline deadlineTask = new Deadline(name, deadlineDate);
+            taskList[taskCounter] = deadlineTask;
+            taskCounter += 1;
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Got it. I've added this task:");
+            System.out.println(deadlineTask.toString());
+            System.out.println("Now you have " + taskCounter + " tasks in the list.");
+            System.out.println(HORIZONTAL_LINE);
+            break;
+        case "event":
+            String eventDate = name.substring(name.indexOf(divider) + 3, name.length());
```

Comment:
Avoid complicated expressions. Consider breaking down the logic into seperate steps to make the code more understandable.